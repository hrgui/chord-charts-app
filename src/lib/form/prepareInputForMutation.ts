export function prepareInputForMutation(input) {
  if (typeof input !== "object" || !input) {
    return input;
  }
  const { id, __typename, ...other } = input;
  const res = {};

  Object.keys(other).forEach(otherKey => {
    let value = other[otherKey];
    if (typeof value === "object" && !Array.isArray(value)) {
      res[otherKey] = prepareInputForMutation(value);
      return;
    }
    if (Array.isArray(value)) {
      value = value.map(v => prepareInputForMutation(v));
    }
    res[otherKey] = value;
  });

  return res;
}

export default prepareInputForMutation;
