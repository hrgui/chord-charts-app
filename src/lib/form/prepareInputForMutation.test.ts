import prepareInputForMutation from "./prepareInputForMutation";

test("prepareInputForMutation", () => {
  expect(
    prepareInputForMutation({
      id: 1,
      __typename: 2,
      name: "Test",
      sections: [
        {
          name: "test",
          id: 1,
          __typename: "Hello"
        }
      ]
    })
  ).toMatchSnapshot();
});
