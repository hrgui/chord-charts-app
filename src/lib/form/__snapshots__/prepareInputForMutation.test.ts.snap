// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepareInputForMutation 1`] = `
Object {
  "name": "Test",
  "sections": Array [
    Object {
      "name": "test",
    },
  ],
}
`;
